types.ts

```diff
@@ -81,6 +81,13 @@ export enum AuthenticationType {
    * @see {@link OAuth2Authentication}
    */
   OAuth2 = 'OAuth2',
+  /**
+   * Authenticate using OAuth2 client credentials.
+   * The API must use a (largely) standards-compliant implementation of OAuth2.
+   *
```

Comment:
For all these comments I mostly just copy/pasted from the OAuth 2 docs so let me know if I should change anything

Reply:
Above I would say something like "Authenticate using the OAuth2 client credentials flow. This is a less common type of OAuth2, which involves exchanging a client ID and secret for a temporary access token."

Likewise, I'd updated the `OAuth2` entry above with something like "Authenticate using the OAuth2 authorization code flow. This is the most common type of OAuth2, which involves the user approving access to their account before being granted a token."